### PR TITLE
Penpal testing runtime

### PR DESCRIPTION
**Pull request by gilescope**
Fixes https://github.com/paritytech/cumulus/issues/1248

### CODE REVIEW

**Comment by gilescope** [comment]
Have a look at the "[changes required to get penpal running](https://github.com/paritytech/cumulus/pull/1254/commits/fc73cea4e194110937e0db9cf9ea21e5b28f7e75)" - maybe there's a slightly nicer way to do it?

File: polkadot-parachain/src/command.rs [hunk 1]
```diff
@@ -85,7 +85,7 @@ fn runtime(id: &str) -> Runtime {
 	} else if id.starts_with("westmint") {
 		Runtime::Westmint
 	} else if id.starts_with("penpal") {
-		Runtime::Penpal(para_id.unwrap())
+		Runtime::Penpal(para_id.unwrap_or(ParaId::new(0)))
```
Comment (gilescope) [inline polkadot-parachain/src/command.rs:88]: At the point where this is used downstream the para id is not used as it's already been parsed and stripped off earlier.

File: polkadot-parachain/src/command.rs [hunk 2]
```diff
@@ -142,8 +146,17 @@ fn load_spec(id: &str) -> std::result::Result<Box<dyn ChainSpec>, String> {
...
+			para_id.expect("Must specify parachain id"),
+			"kusama-local",
+		)),
+		"penpal-polkadot" => Box::new(chain_spec::penpal::get_penpal_chain_spec(
+			para_id.expect("Must specify parachain id"),
+			"polkadot-local",
+		)),
 		// -- Loading a specific spec from disk
 		path => {
+			println!("could not find a path {}", path);
```
Comment (gilescope) [inline polkadot-parachain/src/command.rs:159]:
```suggestion
```

Comment (gilescope) [reply polkadot-parachain/src/command.rs:159]: any stdout println messes up generation of raw json because we pipe output to a file

**Review by NachoPal — APPROVED** [review]

File: polkadot-parachain/src/chain_spec/statemint.rs [hunk 3]
```diff
@@ -19,6 +19,538 @@ use crate::chain_spec::{
 };
 use cumulus_primitives_core::ParaId;
 use hex_literal::hex;
+// <<<<<<< HEAD:polkadot-parachain/src/chain_spec.rs
```
Comment (joepetrowski) [inline polkadot-parachain/src/chain_spec/statemint.rs:22]: some unresolved merge conflict?

Comment (gilescope) [reply polkadot-parachain/src/chain_spec/statemint.rs:22]: Good spot. Yes don't need these comments any more.

File: parachains/runtimes/testing/penpal/src/weights/block_weights.rs [hunk 4]
```diff
@@ -0,0 +1,46 @@
+// This file is part of Substrate.
```
Comment (muharem) [inline parachains/runtimes/testing/penpal/src/weights/block_weights.rs:1]: Should part of 'Cumulus' probably?
Same in all files under 'weights' dir

File: parachains/runtimes/testing/penpal/src/xcm_config.rs [hunk 5]
```diff
@@ -0,0 +1,355 @@
+use super::{
```
Comment (muharem) [inline parachains/runtimes/testing/penpal/src/xcm_config.rs:1]: Some files like this one missing general comment with licence info.

File: parachains/runtimes/testing/penpal/src/weights/block_weights.rs (hunk 4 above)
Comment (gilescope) [reply parachains/runtimes/testing/penpal/src/weights/block_weights.rs:1]: They all say that even for statemine. One would need to parameterise the generator if we really wanted to change that.

File: parachains/runtimes/testing/penpal/src/xcm_config.rs (hunk 5 above)
Comment (gilescope) [reply parachains/runtimes/testing/penpal/src/xcm_config.rs:24]: good spot. Have fixed that and added a few module level doc lines.

**Review by muharem — APPROVED** [review]

**Review by joepetrowski — APPROVED** [review]

**Comment by joepetrowski** [comment]
bot merge

**Pull request merged**